main/lsp/updates.cc

```diff
@@ -378,9 +378,11 @@ LSPLoop::QueryRun LSPLoop::runQuery(unique_ptr<core::GlobalState> gs, const core
     ENFORCE(gs->lspQuery.isEmpty());
     gs->lspQuery = q;
     auto resolved = pipeline::incrementalResolve(*gs, move(updatedIndexed), opts);
+    // pipeline::typecheck must be run before tryApplyDefLocSaver, because pipeline::typecheck runs flatten, which
+    // introduces the invariant that ivars defined directly on a class are put into <static-init>.
+    pipeline::typecheck(gs, move(resolved), opts, workers);
```

Comment:
You're `move`-ing `resolved` here, and then later using `resolved`. This is an error.

I think you have to keep typechecking where it is and call flatten explicitly to transform the AST.

Reply:
Actually, would `resolved = pipeline::typecheck(...)` work?

Reply:
dang, i should've been more careful with those moves. i'll try playing around with it.

Reply:
it seems like adding `resolved = ...` fixes the errors with moving, but causes other tests to fail. we may be depending on those try* functions running before typecheck?